api/src/DataPersister/CampDataPersister.php

```diff
@@ -23,13 +23,20 @@ public function __construct(
         );
     }
 
+    /**
+     * @param Camp $data
+     */
     public function beforeCreate($data): BaseEntity {
         /** @var User $user */
         $user = $this->security->getUser();
         $data->creator = $user;
         $data->owner = $user;
 
-        // TODO prototype cloning logic here? Or in a separate endpoint?
+        // copy from prototype, if given
+        if (isset($data->campPrototypeId)) {
+            $campPrototype = $this->em->find(Camp::class, $data->campPrototypeId);
```

Comment:
Ich glaube das ist eine Sicherheitslücke, solange #2362 nicht implementiert ist: Ich kann so, wenn ich die ID eines beliebigen Lagers kenne auf das ich keinen Zugriff habe, ein neues Lager von diesem klonen und so an die Informationen im Lager herankommen.
Wenn wir davon ausgehen dass #2362 noch vor dem MVP implementiert wird, und wir dann hier auch eine Anpassung inkl. Test machen dass man nicht mehr beliebige Lager klonen kann, dann können wir das von mir aus jetzt so stehen lassen.

Eleganter fände ich hingegen, wenn `campPrototype` eine echte optionale Datenbankrelation wäre (oder wenigstens eine ApiProperty vom Typ `Camp`, sie müsste nicht einmal persistiert werden). Dann müsste das Frontend nämlich konsistent auch hier eine IRI mitsenden, und API Platform würde das Camp dann automatisch für uns finden und dabei auch den Entity Filter berücksichtigen. Ausserdem könnten wir dann mit einer einfachen Validierung sicherstellen, dass das Camp als Prototype zugelassen ist.

Reply:
Das ist mir beim Review auch durch den Kopf gegangen. Aber ist es nicht jetzt schon so, dass `find()` fehlschlägt, wenn ich keinen Zugriff auf das Camp habe? Etwas difficiler wird es, wenn ich in einem Camp eingeladen bin aber noch nicht akzeptiert habe. Dann finde ich das Camp, dürfte den Inhalt aber eigentlich noch nicht sehen.

Deinen Vorschlag finde ich elegant, `campPrototype` als `ApiProperty` zu implementieren. Dann könnte man dort auch eine Validierung draufsetzen, z.B. `Assert\CanCopyFromCamp`. Diese Validierung kann in einem ersten Schritt prüfen, of das Camp ein Prototype ist und später erweitert werden, wenn wir auch von alten Lagern kopieren wollen.

Einen Test dazu wäre in jedem Fall lohnenswert.

Reply:
Der EntityFilter ist via eine DoctrineExtension realisiert. Diese DoctrineExtensions sind ein Konstrukt von API Platform und werden einfach in den eingebauten DoctrineDataProviders (für [Collections](https://github.com/api-platform/core/blob/main/src/Core/Bridge/Doctrine/Orm/CollectionDataProvider.php#L68) und für [Items](https://github.com/api-platform/core/blob/main/src/Core/Bridge/Doctrine/Orm/ItemDataProvider.php#L106)) angesogen, um den generierten Item- / Collection-Query zu ergänzen. Ich glaube nicht dass der EntityManager von Doctrine sich damit herumschlägt.

Reply:
ApiProperty campProperty ist umgesetzt;
Auswahl bei CampCreate.vue zeigt aktuell alle eigens sichtbare Camps (nicht nur Prototypen).
Folgt in neuem PR.